daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/LayeredSequence.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.grammar.primitives
+
+import org.apache.daffodil.grammar.Terminal
+import org.apache.daffodil.dsom._
+import org.apache.daffodil.processors.parsers.{ Parser => DaffodilParser }
+import org.apache.daffodil.processors.unparsers.{ Unparser => DaffodilUnparser }
+import org.apache.daffodil.grammar.Gram
+import org.apache.daffodil.grammar.EmptyGram
+import org.apache.daffodil.util.Misc
+import org.apache.daffodil.processors.parsers.LayeredSequenceParser
+import org.apache.daffodil.processors.unparsers.LayeredSequenceUnparser
+
+case class LayeredSequence(sq: SequenceTermBase, rawTerms: Seq[Gram])
+  extends Terminal(sq, true) {
```

Comment:
The nest you give of LayeredSequence(this, SequenceCombinator(this, terms)) actually has different meaning. For the same reason that it is undesirable to put layering properties on the same sequence that is also carrying other properties. It becomes ambiguous which happens first: 
E.g., in 
{code}
<sequence dfdl:layerTransform="base64_MIME" dfdl:initiator="foo:"... >
{code}
Do we decode base64 first, then look for the initiator, or the other way round? By requiring a separate sequence things are 100% unambiguous, so I'd rather keep things this way for now.

Reply:
Further thought, and I'll put this in scaladoc, which is that while LayeredSequence looks same as sequence now, once SequenceCombinator evolves (as expected), they will differ much more substantially. LayeredSequence will still take only a single body parser, not a vector thereof.

I will change grammar and prim to take single body grammar object, not vector thereof, which will clarify a few things. The grammar will also not be wrapping things that lay down separators around, only to (hopefully) have them optimize out - this is what might not have been working due to inherited non-layer properties on the sequence. It will also prevent an array from being the direct child of a layer, since an array may need separators, etc.